solutions/USACO dec15-high-card-low-card-gold.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: usaco-573
+source: USACO Gold 2015 December
+title: High Card Low Card (Gold)
+author: Qi Wang
+---
+
+**Time Complexity:** $\mathcal{O}(N\logN)$
+
+[Official Analysis](http://www.usaco.org/current/data/sol_cardgame_gold_dec15.html)
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+
+import java.util.*;
+import java.io.*;
+
+public class highcardgold {
+	static int[] C;
+	static List<Integer> B = new ArrayList<>();
+	static List<Integer> F = new ArrayList<>();
+	static List<Integer> S = new ArrayList<>();
+	static int N;
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("cardgame");
+		N = io.nextInt();
+
+		C = new int[2*N+1];
+
+		for (int i = 0; i < N; i++) {
+			int c = io.nextInt();
+			C[c]++;
+			if(i < N/2)
+				F.add(c);
+			else
+				S.add(c);
+		}
+
+		for (int i = 1; i < 2 * N + 1; i++) {
+			if(C[i] == 0) B.add(i);
+		}
+
+		Collections.sort(F);
+		Collections.sort(B);
```

Comment:
Technically redundant (B is already sorted).

Reply:
true